### PR TITLE
Decoupling account confirmation and hacker creation

### PR DESCRIPTION
**Pull request by pashakondratyev**

### CODE REVIEW

File: middlewares/hacker.middleware.js [hunk 1]
```diff
@@ -74,6 +74,27 @@ function addDefaultStatus(req, res, next) {
...
 }
 
+/**
+ * Verifies that account is confirmed and of proper type from the account ID passed in req.body.accountId
+ * @param {{body: {accountId: ObjectId}}} req 
+ * @param {*} res 
+ * @param {(err?) => void} next 
+ */
+async function validateConfirmedStatus(req, res, next) {
+    Services.Account.findById(req.body.accountId).then(
```
Comment (pierreTklein) [inline middlewares/hacker.middleware.js]: Convert to await

File: routes/api/auth.js [hunk 2]
```diff
@@ -145,6 +146,29 @@ module.exports = {
...
+         * @apiGroup Authentication
+         * @apiVersion 0.0.8
+         *
+         * @apiParam {String} JWT for confirming the account
+         *
+         * @apiSuccess {string} message Success message
+         * @apiSuccess {object} data empty
+         * @apiSuccessExample {json} Success-Response:
+         *      {"message": "Successfully confirmed account", "data": {}}
+         *
```
Comment (pierreTklein) [inline routes/api/auth.js]: Is there a fail case

File: tests/account.test.js [hunk 3]
```diff
@@ -103,6 +105,18 @@ describe("POST create account", function () {
     });
 });
 
+describe("POST confirm account", function () {
+    it("should SUCCEED and confirm the account", function(done) {
+        chai.request(server.app)
+        .post('/api/auth/confirm/' + confirmationToken)
+        .type("application/json")
+        .end(function (err, res){
+            res.should.have.status(200);
```
Comment (pierreTklein) [inline tests/account.test.js]: Check result body

File: middlewares/auth.middleware.js [hunk 4]
```diff
@@ -207,6 +207,7 @@ async function validateConfirmationToken(req, res, next) {
     const userObj = await Services.Account.findById(req.body.decodedToken.accountId);
     if (confirmationObj && userObj && (confirmationObj.accountId == userObj.id)) {
         userObj.confirmed = true;
+        userObj.accountType = req.body.decodedToken.accountType;
```
[the remark this answers is not in the document]
Comment (pashakondratyev) [reply middlewares/auth.middleware.js]: Yes

[the remark this answers is not in the document]
Comment (pashakondratyev) [reply middlewares/auth.middleware.js]: Just kidding, fixed

**Comment by pierreTklein** [comment]
Resolve the conflicts tho

**Comment by pashakondratyev** [comment]
>Resolve the conflicts tho

@pierreTklein Yeah was working on that before class, will get it in tonight

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Pull request merged**